### ISSUE TITLE
Uncaught exception when importing CSV

### ISSUE THREAD

**Issue by nathanosdev** [bandrewk/nexotransactionanalyzer#2]
First off, thanks for working on this tool! I've been using spreadsheets up until now to track my progress on Nexo.

I'm importing a newly exported CSV file that has been massaged to work as per your instructions on Reddit: https://www.reddit.com/r/Nexo/comments/syjf95/nexo_transaction_analyzer_beta/

> For example:
>\<transaction id>,Exchange,EURX/LINK,LINK 11.02630541,$123,approved / Exchange EURX to Chainlink,$0.00,\<date>
>should be:
>\<transaction id>,Exchange,EURX/LINK,-500.00 / 11.02630541,$123,approved / Exchange EURX to Chainlink,$0.00,\<date>

After importing the file, the app navigates to the "Overview" but all values are 0, except "Transactions" which is 1. There's 143 transactions in the imported CSV file. The exception in the console is this:
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'stop')
    at CApp.#FreeParticleSystems (app.js:581:15)
    at CApp.ReceiveCurrentExchangeRates (app.js:333:30)
    at statistics.js:278:29
```

**Comment by bandrewk** [comment]
I just realized I had notifications being sent into Nirvana - sorry for the delay. 

- What browser + version are you running on?
- Do you have unsupported transactions (see readme.md for a list of supported ones)?
- Does any error pop up in the console when you visit the website without interacting?
- Are you able to pinpoint the error to a specific line in your transactions? (e.g. start with first 100 entries and increase till it stops working)

Here's what one of my fixed `EXCHANGE` transactions look like:

![image](https://user-images.githubusercontent.com/6756767/158146764-270230d6-b456-419d-8770-a6be799b321c.png)
_The white space is important  too._

**Comment by nathanosdev** [comment]
Sorry to skip all your questions, but it's not related to the transaction import - there is a race condition in the code.

The problem is this `StartLoadingAnimation` function that works asynchronously. 
https://github.com/bandrewk/nexotransactionanalyzer/blob/main/js/app.js#L585

This animation finishes loading _after_  `ReceiveCurrentExchangeRates` is called here:
https://github.com/bandrewk/nexotransactionanalyzer/blob/main/js/app.js#L311

So that results in the exception that I posted above happening here:
https://github.com/bandrewk/nexotransactionanalyzer/blob/main/js/app.js#L580-L582

I was able to fix it locally by wrapping the 2nd `particles.stop()` call in an if condition:
```js
 if (particles) {
  particles.stop();
  particles.destroy();
}
```

Maybe you'll want to find a cleaner solution that this though.
